python/pyiceberg/cli/console.py

```diff
@@ -0,0 +1,228 @@
+        ctx.obj["catalog"] = RestCatalog(name=catalog, properties={}, uri=uri, credential=credential)
+    elif uri.startswith("thrift"):
+        ctx.obj["catalog"] = HiveCatalog(name=catalog, properties={}, uri=uri)
+    else:
+        raise ValueError("Could not determine catalog type from uri. REST (http/https) and Hive (thrift) is supported")
+
+
+def _get_catalog_and_output(ctx) -> Tuple[Catalog, Output]:
+    """
+    Small helper to set the types
+    """
+    return ctx.obj["catalog"], ctx.obj["output"]
+
+
+@run.command()
+@click.pass_context
+@click.argument("parent", required=False)
+def list(ctx, parent: Optional[str]):
+    catalog, output = _get_catalog_and_output(ctx)
+
+    # still wip, will become more beautiful
+    # https://github.com/apache/iceberg/pull/5467/
+    # has been merged
+    if parent:
+        identifiers = catalog.list_tables(parent)
+    else:
+        identifiers = catalog.list_namespaces()
+
+    output.identifiers(identifiers)
+
+
+@run.command()
+@click.argument("identifier")
+@click.pass_context
+def describe(ctx, identifier: str):
+    """Describes a table"""
+    catalog, output = _get_catalog_and_output(ctx)
+    identifier_tuple = Catalog.identifier_to_tuple(identifier)
+
+    if len(identifier_tuple) > 1:
+        try:
+            table = catalog.load_table(identifier)
+            output.describe_table(table)
+        except NoSuchNamespaceError as exc:
+            output.exception(exc)
+    else:
+        output.exception(NoSuchTableError(f"Expected a namespace, got: {identifier}"))
+
+
+@run.command()
+@click.option("--type", type=click.Choice(["any", "namespace", "table"]), default="any")
+@click.argument("identifier")
+@click.pass_context
+def properties(ctx, type: Literal["name", "namespace", "table"], identifier: str):
+    """Fetches the properties from the namespace/table"""
+    catalog, output = _get_catalog_and_output(ctx)
+    identifier_tuple = Catalog.identifier_to_tuple(identifier)
+
+    found_namespace = False
+    try:
+        namespace_properties = catalog.load_namespace_properties(identifier_tuple)
+        output.describe_properties(namespace_properties)
+        found_namespace = True
+    except NoSuchNamespaceError as exc:
+        if type != "any":
+            output.exception(exc)
+
+    found_table = False
+    # We expect a namespace
+    if len(identifier_tuple) > 1:
+        try:
+            table = catalog.load_table(identifier_tuple)
+            assert table.metadata
+            output.describe_properties(table.metadata.properties)
+            found_table = True
+        except NoSuchTableError as exc:
+            if type != "any":
+                output.exception(exc)
+
+    if type == "any" and not found_namespace and not found_table:
+        output.exception(NoSuchNamespaceError(f"Could not find table/namespace with identifier {identifier}"))
+
+
+@run.command()
+@click.argument("identifier")
+@click.pass_context
+def schema(ctx, identifier: str):
```

Comment:
I accidentally hit the single part table name here as well. We may want to add `with load(identifier) as table:` to encapsulate the `len(identifier_tuple) > 1` check and other handling above.